algos/eq/equalizers_tuning.rst

```diff
@@ -0,0 +1,924 @@
+design for any finite time impulse response / frequency response is
+simple and robust. The filters do not oscillate by design so the
+rounding errors do not appear as noise. The rounding of coefficients
+into a fixed word length only impairs slightly the response but the
+effect can be usually ignored. Therefore the FIR equalizers especially
+when used with 24 and 32 bit audio format are compatible with studio
+like 24 bit audio quality.
+
+Due finite response (often limited by DSP resources) the FIR filters
+are not practical for lowest frequencies unless very long filters are
+used. The longer the filter is the more DSP RAM and MCPS the
+processing consumes. However FIR filters are great for mid and high
+frequencies equalization. The next example equalizes those frequencies
+for the previously done measurement.
+
+The initial script for tuning is shown below. Alternatively for other
+equipment the data import could be done in Excel format and use
+function xlsread(); to read a matrix and then extract the frequency
+and magnitude columns.
+
+.. code-block:: octave
+
+		%% Load measurement data, variable f and m
+		load example_dut.mat;
+
+		%%  EQ settings
+		eq1 = eq_defaults();         % Get defaults
+		eq1.fs = 48e3;               % Set sample rate
+		eq1.norm_type = 'loudness';  % Normalize criteria can be loudness/peak/1k
+		eq1.norm_offs_db = -3;       % Offset in dB to normalize, -3dB loudness
+		eq1.logsmooth_plot = 1.0;    % Smooth over 1.0 octaves
+		eq1.logsmooth_eq = 1.0;      % Smooth over 1.0 octaves
+		eq1.enable_fir = 1;          % By default both FIR and IIR disabled
+		eq1.fir_beta = 3.0;          % Lower beta is more accurate but be careful
+		eq1.fir_length = 90;         % Minimize this vs. fmin/fmax choice
+		eq1.fir_autoband = 0;        % Select manually frequency limits
+		eq1.fmin_fir = 700;          % Equalization starts from 800 Hz
+		eq1.fmax_fir = 13e3;         % Equalization ends at 13 kHz
+		eq1.fir_minph = 1;           % Check result carefully if 1 is used, 0 is safe
+		eq2 = eq1;                   % Copy settings to second EQ
+
+		%% Design left channel EQ
+		eq1.raw_f = f;               % Measurement Hz
+		eq1.raw_m_db = m(:,1);       % Measurement dB, left ch
+		eq1 = eq_compute(eq1);
+		eq_plot(eq1, 10);
+
+The run of this script creates these plots. Note the choice of 1.0
+octaves smoothing for both plotting and EQ target derivation. It’s
+best to start carefully with such a high amount of smoothing to avoid
+to equalize highly uncertain details of frequency response.
+
+The smoothed version of the response becomes very flat in the
+equalized version. However the simulated raw response still contains a
+lot of ripple especially at high bands. It’s an industry standard to
+use ⅓ octaves smoothing since it quite well matches human ear
+psycho-acoustics. Therefore ⅓ octaves should be the smallest feasible
+width of octaves smoothing to use.
+
+.. figure:: Picture_response_with_smoothing.png
```

Comment:
It is recommended to use SVG images because 1) they are scalable without loosing a quality and 2) text labels etc. that are part of the graphics are searchable.

Reply:
Yep, octave seems to now have svg as option for print. Hopefully also web browsers are OK with it. I will test it for next contribution.

Reply:
All plantuml diagrams are already rendered to SVG and I have not seen any issues with web browsers reported.